utils.py

```diff
@@ -493,8 +552,80 @@ def cli_download(uri, destination):
+            return
+
+def set_appimage_symlink(selected_appimage=config.WINE64_APPIMAGE_FILENAME):
+    if selected_appimage is None:
+        logging.warning(f"No AppImage provided.")
+        return
+
+    appimage_symlink_path = f"{config.APPDIR_BINDIR}/{config.APPIMAGE_LINK_SELECTION_NAME}"
+    if os.path.lexists(appimage_symlink_path) and not os.path.exists(os.readlink(appimage_symlink_path)):
+        # We must call this separately from the test below as Python does not recognize a broken symlink as existing. Both tests should cover corner cases.
+        delete_symlink(appimage_symlink_path)
+    elif os.path.exists(appimage_symlink_path):
+        delete_symlink(appimage_symlink_path)
+
+    appimage_filename = os.path.basename(selected_appimage)
+    appimage_filepath = f"{config.APPDIR_BINDIR}/{appimage_filename}"
+
+    def store_appimage():
+        logging.info(f"Copying AppImage to {config.APPDIR_BINDIR}.")
+        shutil.copy(selected_appimage, f"{config.APPDIR_BINDIR}")
+        os.symlink(appimage_filename, appimage_symlink_path)
+        config.WINE64_APPIMAGE_FILENAME = appimage_filename
+
+    copy_message = f"Should the program copy the selected AppImage to the {config.APPDIR_BINDIR} directory?"
+
+    # Determine if user wants their AppImage in the Logos on Linux bin dir.
+    if os.path.exists(appimage_filepath):
+        confirm = False
+    else:
+        if config.DIALOG == "tk":
+            tk_root = tk.Tk()
+            tk_root.withdraw()
+            confirm = messagebox.askquestion("Confirmation", copy_message)
+            tk_root.destroy()
+            if confirm == 'yes':
+                store_appimage()
+        elif config.DIALOG == "curses":
+            confirm = curses_confirm("Confirmation", copy_message)
+            if confirm == True:
+                store_appimage()
+        else:
+            confirm = msg.cli_question(copy_message)
+            if confirm == True:
+                store_appimage()
+
+    # If not, user the selected AppImage's full path for link creation.
+    if confirm == False or confirm == 'no':
+        if not os.path.isabs(selected_appimage):
+            raise ValueError("Selected AppImage path must be an absolute path.")
+        os.symlink(selected_appimage, appimage_symlink_path)
+        config.WINE64_APPIMAGE_FILENAME = selected_appimage
+
+def find_appimage_files(directory=""):
+    appimages = []
+    directories = [config.APPDIR_BINDIR, get_user_downloads_dir(), directory]
+    for dir in directories:
+        for root, dirs, files in os.walk(dir):
+            for filename in files:
+                if filename.lower().startswith("wine") and filename.lower().endswith("appimage"):
+                    appimage_path = os.path.join(root, filename)
+                    if os.access(appimage_path, os.X_OK):
+                        appimages.append(appimage_path)
+                    else:
+                        pass
```

Comment:
`pathlib.Path()` is very compact and efficient when it comes to recursive searching. You could do something like:
```python
        appimage_paths = Path(dir).rglob('wine*.appimage', case_sensitive=False)
        for p in appimage_paths:
            if os.access(p, os.X_OK):
                appimages.append(p)
```